### PR TITLE
fix file input rerendering behaviour

### PR DESCRIPTION
**Pull request by maxb94**

### CODE REVIEW

File: src/components/molecules/ImageInput/ImageCollectionInput.tsx [hunk 1]
```diff
@@ -57,6 +54,7 @@ export const ImageCollectionInput: React.FC<ImageInputProps> = (props) => {
     string | undefined
   >(imageCollection.find((url) => url === imageUrlFromAPI));
   const [imageFiles, setImageFiles] = useState<FileList>();
+  console.log("imageFiles", imageFiles);
```
Comment (thecadams) [inline src/components/molecules/ImageInput/ImageCollectionInput.tsx:57]: nit: looks like this was left in

File: src/components/molecules/ImageInput/ImageCollectionInput.tsx [hunk 2]
```diff
@@ -73,26 +71,25 @@ export const ImageCollectionInput: React.FC<ImageInputProps> = (props) => {
...
     },
     []
   );
 
   // this keeps the component state synchronised with the parent form state
   useEffect(() => {
     if (selectedCollectionImageUrl) {
-      setImageFileRef.current(undefined);
-      setImageUrlRef.current(selectedCollectionImageUrl);
+      console.log(`${fieldName} 1`);
```
Comment (thecadams) [inline src/components/molecules/ImageInput/ImageCollectionInput.tsx:82]: And this one

File: src/components/molecules/ImageInput/ImageCollectionInput.tsx [hunk 3]
```diff
@@ -145,6 +142,7 @@ export const ImageCollectionInput: React.FC<ImageInputProps> = (props) => {
         }}
       >
         <input
+          key={fileInputKey.toString()} // trick react to recreate the file input
```
Comment (thecadams) [inline src/components/molecules/ImageInput/ImageCollectionInput.tsx:145]: You able to leave a slightly more detailed comment? Trying to understand how forcing re render affects things

File: src/components/molecules/ImageInput/ImageCollectionInput.tsx [hunk 4]
```diff
@@ -73,26 +71,25 @@ export const ImageCollectionInput: React.FC<ImageInputProps> = (props) => {
...
-      setImageFileRef.current(undefined);
-      setImageUrlRef.current(selectedCollectionImageUrl);
+      console.log(`${fieldName} 1`);
+      setValue(`${fieldName}File`, undefined, false);
+      setValue(`${fieldName}Url`, selectedCollectionImageUrl, false);
       setImageUrlForPreview(selectedCollectionImageUrl);
     } else if (imageFiles && imageFiles.length > 0) {
-      setImageUrlRef.current(undefined);
-      setImageFileRef.current(imageFiles);
+      console.log(`${fieldName} 2`);
```
Comment (thecadams) [inline src/components/molecules/ImageInput/ImageCollectionInput.tsx:87]: And a third one

Comment (sintinilorenzo) [inline src/components/molecules/ImageInput/ImageCollectionInput.tsx:87]: get rid of console.log

File: src/components/molecules/ImageInput/ImageCollectionInput.tsx (hunk 2 above)
Comment (sintinilorenzo) [inline src/components/molecules/ImageInput/ImageCollectionInput.tsx:82]: get rid of line

File: src/components/molecules/ImageInput/ImageCollectionInput.tsx [hunk 5]
```diff
@@ -73,26 +71,25 @@ export const ImageCollectionInput: React.FC<ImageInputProps> = (props) => {
...
   );
 
   // this keeps the component state synchronised with the parent form state
   useEffect(() => {
     if (selectedCollectionImageUrl) {
-      setImageFileRef.current(undefined);
-      setImageUrlRef.current(selectedCollectionImageUrl);
+      console.log(`${fieldName} 1`);
+      setValue(`${fieldName}File`, undefined, false);
+      setValue(`${fieldName}Url`, selectedCollectionImageUrl, false);
```
Comment (sintinilorenzo) [inline src/components/molecules/ImageInput/ImageCollectionInput.tsx:84]: comment test

File: src/components/molecules/ImageInput/ImageCollectionInput.tsx (hunk 3 above)
Comment (maxb94) [reply src/components/molecules/ImageInput/ImageCollectionInput.tsx:142]: sure

**Pull request merged**